src/main/scala/cromwell/engine/WorkflowManager.scala

```diff
@@ -25,20 +30,33 @@ trait WorkflowManager {
 
   val backend: Backend
 
-  def generateWorkflow(id: WorkflowId, wdl: WdlSource, inputs: WorkflowInputs): Try[Workflow]
-  def workflowStatus(id: WorkflowId): Option[WorkflowState]
+  def generateWorkflow(id: WorkflowId, wdl: WdlSource, inputs: binding.WorkflowInputs): Try[Workflow]
+
+  def workflowStatus(id: WorkflowId): Option[WorkflowState] = workflowStates.get(id)
+  def workflowOutputs(id: WorkflowId): Future[Option[binding.WorkflowOutputs]]
+  def submitWorkflow(wdl: WdlSource, inputs: binding.WorkflowInputs): Future[Try[WorkflowId]]
+  def updateWorkflowState(workflow: Workflow, state: WorkflowState): Unit
 
-  private val workflowStore = new WriteOnceStore[WorkflowId, Workflow]
+  def idByWorkflow(workflow: Workflow): Option[WorkflowId] = {
+    workflowStore.toMap find {case (k, v) => v == workflow} map {_._1}
```

Comment:
Oops, put this in the wrong place. Replaced with workflowStore.toMap collectFirst {case (k, v) if v == workflow => k}